spec/lib/efile/nj/nj1040_spec.rb

```diff
@@ -0,0 +1,144 @@
+        end
+
+        it 'calculates total line 7 exemption amount correctly' do
+          expect(instance.lines[:NJ1040_LINE_7].value).to eq(0)
+        end
+      end
+    end
+
+    context 'when filer has a spouse,' do
+      let(:intake) { create(:state_file_nj_intake, :married) }
+      context 'when filer is older than 65 and spouse is older than 65' do
+        let(:intake) { create(:state_file_nj_intake, :married_spouse_over_65, :self_over_65) }
+        before do
+          instance.calculate
+        end
+        it 'sets the self checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SELF].value).to eq(true)
+        end
+
+        it 'sets the spouse checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SPOUSE].value).to eq(true)
+        end
+
+        it 'calculates total line 7 exemption amount correctly' do
+          expect(instance.lines[:NJ1040_LINE_7].value).to eq(2000)
+        end
+      end
+
+      context 'when filer is younger than 65 and spouse is older than 65' do
+        let(:intake) { create(:state_file_nj_intake, :married_spouse_over_65) }
+        before do
+          instance.calculate
+        end
+        it 'sets the self checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SELF].value).to eq(false)
+        end
+
+        it 'sets the spouse checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SPOUSE].value).to eq(true)
+        end
+
+        it 'calculates total line 7 exemption amount correctly' do
+          expect(instance.lines[:NJ1040_LINE_7].value).to eq(1000)
+        end
+      end
+
+      context 'when filer is older than 65 and spouse is younger than 65' do
+        let(:intake) { create(:state_file_nj_intake, :married, :self_over_65) }
+        before do
+          instance.calculate
+        end
+        it 'sets the self checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SELF].value).to eq(true)
+        end
+
+        it 'sets the spouse checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SPOUSE].value).to eq(false)
+        end
+
+        it 'calculates total line 7 exemption amount correctly' do
+          expect(instance.lines[:NJ1040_LINE_7].value).to eq(1000)
+        end
+      end
+
+      context 'when filer is younger than 65 and spouse is younger than 65' do
+        let(:intake) { create(:state_file_nj_intake, :married) }
+        before do
+          instance.calculate
+        end
+        it 'sets the self checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SELF].value).to eq(false)
+        end
+
+        it 'sets the spouse checkbox correctly' do
+          expect(instance.lines[:NJ1040_LINE_7_SPOUSE].value).to eq(false)
+        end
+
+        it 'calculates total line 7 exemption amount correctly' do
+          expect(instance.lines[:NJ1040_LINE_7].value).to eq(0)
+        end
+      end
+    end
+  end
+end
```

Comment:
I think my general comments for this file is that the structure could look like this 

Efile::Nj::Nj1040 do
  context when single
   it does "bla bla what you expect it to do"
      expect line 6 to equal x
      expect line 7 to equal y
      expect line 7seld to equal 
      etc etc
   end 
  end

  context when married
   it does "bla bla what you expect it to do"
      expect line 6 to equal x
      expect line 7 to equal y
      expect line 7seld to equal 
      etc etc
   end 
 context when spouse is older than 65
  end

context when primary older than 65
end

context when primary is younger than 65
end

end

... I might be missing some things but let me know if this doesn't make sense

Reply:
Can you clarify how this would test the different married situations? 

In my mind there are four cases we need to test when the filer is married: 1T+2T; 1T+2F; 1F+2T; 1F+2F, where T = older and F = younger for primary/spouse respectively. 

Right now in the test block i'm seeing above, I see coverage for each condition individually but not the test cases where they're together.

context 1T
context 2T
context 1F
context 2F

Might be easier to discuss over a call in-person, happy to do so in our next pairing session.